src/read.rs

```diff
@@ -228,6 +230,50 @@ impl<R: Read+io::Seek> ZipArchive<R>
         })
     }
 
+    /// Extract a Zip archive into a directory.
+    ///
+    /// Paths are sanitized so that they cannot escape the given directory.
+    ///
+    /// # Platform-specific behaviour
+    ///
+    /// On unix systems permissions from the zip file are preserved, if they exist.
+    pub fn extract(&mut self, directory: &Path) -> ZipResult<()> {
```

Comment:
I think we should follow the std lib and take `AsRef<Path>` instead of `&Path`. You can see an example [here](https://doc.rust-lang.org/stable/std/path/struct.PathBuf.html#method.push).